src/api/github/checkRun.ts

```diff
@@ -0,0 +1,119 @@
+>;
+
+export type Annotation = NonNullable<Output['annotations']>[number];
+
+const GITHUB_MAX_ANNOTATIONS = 50;
+
+const isGitHubAnnotationsEnabled = (): boolean =>
+  Boolean(
+    process.env.BUILDKITE_BUILD_NUMBER &&
+      process.env.BUILDKITE_COMMIT &&
+      process.env.BUILDKITE_REPO &&
+      process.env.GITHUB_API_TOKEN,
+  );
+// Pulls out the GitHub Owner + Repo String from repo urls eg.
+// git@github.com:seek-oss/skuba.git
+// https://github.com/seek-oss/skuba.git
+// Pulls out `seek-oss` as owner and `skuba` as repo
+const ownerRepoRegex = new RegExp(/github.com(?::|\/)(.*)\/(.*).git/);
+
+const getOwnerRepo = (): { owner: string; repo: string } => {
+  const match = ownerRepoRegex.exec(process.env.BUILDKITE_REPO as string);
+  const owner = match?.[1];
+  const repo = match?.[2];
+
+  if (!owner || !repo) {
+    throw new Error(
+      'Could not extract GitHub owner/repo from BUILDKITE_REPO environment variable',
+    );
+  }
+
+  return {
+    owner,
+    repo,
+  };
+};
+
+/**
+ * Create a uniform title format for our check runs
+ * @param conclusion - `failure` or `success`
+ * @param annotationsLength - Number of annotations added
+ * @returns Title eg. Build #12 failed (24 annotations added)
+ */
+const createTitle = (
+  conclusion: 'failure' | 'success',
+  annotationsLength: number,
+): string => {
+  const build = `Build #${process.env.BUILDKITE_BUILD_NUMBER as string}`;
+  const numAnnotations =
+    annotationsLength > GITHUB_MAX_ANNOTATIONS
+      ? GITHUB_MAX_ANNOTATIONS
+      : annotationsLength;
+  const status = conclusion === 'success' ? 'passed' : 'failed';
+
+  const plural = numAnnotations === 1 ? '' : 's';
+  return `${build} ${status} (${numAnnotations} annotation${plural} added)`;
+};
+
+/**
+ * Adds more context to the summary provided
+ * @param summary - report summary
+ * @param annotationsLength - Number of annotations added
+ * @returns summary with extra metadata
+ */
+const createEnrichedSummary = (
+  summary: string,
+  annotationsLength: number,
+): string =>
+  [
+    summary,
+    ...(annotationsLength > GITHUB_MAX_ANNOTATIONS
+      ? [
+          `There were ${annotationsLength} annotations created. However, the number of annotations displayed has been capped to ${GITHUB_MAX_ANNOTATIONS}`,
+        ]
+      : []),
+  ].join('\n\n');
+
+interface CreateCheckRunParameters {
+  name: string;
+  summary: string;
+  annotations: Annotation[];
+  conclusion: 'failure' | 'success';
+}
+
+export const createCheckRunFromBuildkite = async ({
+  name,
+  summary,
+  annotations,
+  conclusion,
+}: CreateCheckRunParameters): Promise<void> => {
+  if (!isGitHubAnnotationsEnabled()) {
+    return;
+  }
+
+  const client = new Octokit({
+    auth: process.env.GITHUB_API_TOKEN,
+  });
+  const { owner, repo } = getOwnerRepo();
+  const title = createTitle(conclusion, annotations.length);
+  const enrichedSummary = createEnrichedSummary(summary, annotations.length);
+  const checkRunName = `skuba/${name}`;
```

Comment:
IMO this magic doesn't really belong here and we should pass through the `name` parameter verbatim. Users will be able to use this function to write their own arbitrary check runs, and those wouldn't really make sense under a `skuba/` namespace.